src/runtime.rs

```diff
@@ -41,6 +43,13 @@ impl std::fmt::Debug for Revision {
     }
 }
 
+#[derive(Debug)]
+pub(crate) struct RevisionControlSystem {
+    revision: Revision,
+    waker: Waker,
```

Comment:
Can you define a type `RevisionControlState` with these two fields and make a field here like

```
    state: RwLock<RevisionControlState>,
```